front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/formatMargin.ts

```diff
@@ -6,16 +6,18 @@ const formatMargin = (pathSteps: PathStep[]): Margin | undefined => {
   const values: string[] = [];
 
   pathSteps.forEach((step, index) => {
+    const { id, theoreticalMargin } = step;
+
     if (index === 0) {
-      values.push(step.theoreticalMargin || 'none');
+      values.push(theoreticalMargin || 'none');
     } else if (index === pathSteps.length - 1) {
       if (values.length === 1 && values[0] !== 'none') {
-        boundaries.push(step.id);
+        boundaries.push(id);
         values.push('none');
       }
-    } else if (step.theoreticalMargin) {
-      boundaries.push(step.id);
-      values.push(step.theoreticalMargin);
+    } else if (theoreticalMargin) {
+      boundaries.push(id);
+      values.push(theoreticalMargin.startsWith('0') ? 'none' : theoreticalMargin);
```

Comment:
Seems safer to use `=== '0%'` here, I think? The user may enter e.g. `05%`.

Reply:
I thought about that but if we do that we need also to check `=== '0km/100km`

Reply:
Yeah, I think I'd prefer to tighten up the checks and hit editoast errors in weird cases like `000%`, instead of silently changing the value the user entered in weird cases like `05%`.